dipy/reconst/dti.py

```diff
@@ -1680,43 +1691,49 @@ def nlls_fit_tensor(design_matrix, data, weighting=None,
         start_params = ols_params[vox]
         # Do the optimization in this voxel:
         if jac:
-            this_tensor, status = opt.leastsq(_nlls_err_func, start_params,
-                                              args=(design_matrix,
-                                                    flat_data[vox],
-                                                    weighting,
-                                                    sigma),
-                                              Dfun=_nlls_jacobian_func)
+            this_param, status = opt.leastsq(_nlls_err_func, start_params,
+                                             args=(design_matrix,
+                                                   flat_data[vox],
+                                                   weighting,
+                                                   sigma),
+                                             Dfun=_nlls_jacobian_func)
         else:
-            this_tensor, status = opt.leastsq(_nlls_err_func, start_params,
-                                              args=(design_matrix,
-                                                    flat_data[vox],
-                                                    weighting,
-                                                    sigma))
+            this_param, status = opt.leastsq(_nlls_err_func, start_params,
+                                             args=(design_matrix,
+                                                   flat_data[vox],
+                                                   weighting,
+                                                   sigma))
 
-        # The parameters are the evals and the evecs:
+        # Convert diffusion tensor parameters to the evals and the evecs:
         try:
             evals, evecs = decompose_tensor(
-                from_lower_triangular(this_tensor[:6]))
-            dti_params[vox, :3] = evals
-            dti_params[vox, 3:] = evecs.ravel()
+                from_lower_triangular(this_param[:6]))
+            params[vox, :3] = evals
+            params[vox, 3:12] = evecs.ravel()
             if return_S0_hat:
-                model_S0[vox] = np.exp(-this_tensor[6])
+                model_S0[vox] = np.exp(-this_param[-1])
+            if not dti:
```

Comment:
This is basically just DKI, right? The `not dti` seems to suggest there are other models as well, which may be confusing.

Reply:
Also, does this really need to be inside the `try` block? You seem to do this again below in the `except` block.

Reply:
I have in mind to use the same trick for other models based on cumulant expansion (e.g. correlation tensor imaging, future work), that is why I decided to call this variable ´dti´.

Reply:
It was inside try blook because in expect blook another variable is called. Anyway, I will arrange a way to take it out to simplify code.